cmd/server/cmd/server.go

```diff
@@ -408,17 +416,39 @@ func setupServer(sa *serverArgs, info map[string]template.Info, adapters []adptr
 		fatalf("Unable to listen on socket: %v", err)
 	}
 
+	sm := http.NewServeMux()
 	// NOTE: this is a temporary solution for provide bare-bones debug functionality
 	// for mixer. a full design / implementation of self-monitoring and reporting
 	// is coming. that design will include proper coverage of statusz/healthz type
 	// functionality, in addition to how mixer reports its own metrics.
-	http.Handle(metricsPath, promhttp.Handler())
-	http.HandleFunc(versionPath, func(out http.ResponseWriter, req *http.Request) {
+	sm.Handle(metricsPath, promhttp.Handler())
+	sm.HandleFunc(versionPath, func(out http.ResponseWriter, req *http.Request) {
 		if _, verErr := out.Write([]byte(version.Info.String())); verErr != nil {
 			printf("error printing version info: %v", verErr)
 		}
 	})
-	monitoring := &http.Server{Addr: fmt.Sprintf(":%d", sa.monitoringPort)}
+	sm.HandleFunc(configPath, func(out http.ResponseWriter, req *http.Request) {
+		path := req.URL.Path[len(configPath):len(req.URL.Path)]
+		cfg := configGetter.ConfigGet(path)
+		if cfg == nil {
+			out.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		var ba []byte
+		var verr error
+
+		if ba, verr = json.Marshal(cfg); err != nil {
+			printf("error converting %v: %v", cfg, verr)
+			out.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		out.Header().Add("content-type", "application/json")
```

Comment:
can we also allow yaml as for output?  I feel like it's more visible and debuggable for human readers.

Maybe it can be dispatched `Accept` request header.

Reply:
One can pipe the json output thru `jq` to make it more readable.

Reply:
It's okay to be JSON by default. I'm saying to allow yaml when `Accept: application/yaml` header is specified.